src/resources/ble_esp32.cc

```diff
@@ -469,27 +689,30 @@ PRIMITIVE(scan_next) {
 PRIMITIVE(scan_stop) {
   ARGS(BLEResourceGroup, group);
 
-  if (group->is_scanning()) {
+  if (BLEResourceGroup::is_scanning()) {
     int err = ble_gap_disc_cancel();
     if (err != ESP_OK) {
       return Primitive::os_error(err, process);
     }
-    group->set_scanning(false);
   }
 
   return process->program()->null_object();
 }
 
 PRIMITIVE(advertise_start) {
-  ARGS(BLEResourceGroup, group, int64, duration_us, int, interval_us);
+  ARGS(BLEResourceGroup, group, int64, duration_us, int, interval_us, int, conn_mode);
 
-  if (group->is_advertising()) ALREADY_EXISTS;
+  if (BLEResourceGroup::is_advertising()) ALREADY_EXISTS;
 
   int32 duration_ms = duration_us < 0 ? BLE_HS_FOREVER : duration_us / 1000;
 
   struct ble_gap_adv_params adv_params = { 0 };
-  // No support for connections yet.
-  adv_params.conn_mode = BLE_GAP_CONN_MODE_NON;
+  if (group->server_config() != null) {
```

Comment:
This looks weird. You check the server_config and do the same thing if it is null or not.